MekHQ/src/mekhq/campaign/personnel/ranks/Rank.java

```diff
@@ -19,166 +19,212 @@
+                name = split[0];
+                level = Integer.parseInt(split[1].trim());
+            } else {
+                level = 1;
+            }
+            getRankNames().put(profession, name);
+            getRankLevels().put(profession, level);
+        }
+    }
+    //endregion Initialization
+
+    public String getName(final Profession profession) {
+        return getRankNames().get(profession);
     }
 
-    public int getRankLevels(int profession) {
-    	return rankLevels.get(profession);
+    public String getNameWithLevels(final Profession profession) {
+        return getRankNames().get(profession) + ((getRankLevels().get(profession) > 1) ? ":" + getRankLevels().get(profession) : "");
     }
 
-    public String getRankNamesAsString() {
-    	StringJoiner joiner = new StringJoiner(",");
-    	for (String name : rankNames) {
-    		if (rankLevels.size() > 0 && rankLevels.get(rankNames.indexOf(name)) > 0) {
-    			joiner.add(name + rankLevels.get(rankNames.indexOf(name)).toString());
-    		} else {
+    public String getRankNamesAsString(final String delimiter) {
+        StringJoiner joiner = new StringJoiner(delimiter);
+        for (final Profession profession : Profession.values()) {
+            String name = getRankNames().get(profession);
+            name = (name == null) ? "-" : name;
+            if (getRankLevels().containsKey(profession) && (getRankLevels().get(profession) > 1)) {
+                joiner.add(name + getRankLevels().get(profession));
+            } else {
                 joiner.add(name);
             }
-    	}
-    	return joiner.toString();
+        }
+        return joiner.toString();
     }
 
-    public void writeToXml(PrintWriter pw1, int indent) {
-        pw1.println(MekHqXmlUtil.indentStr(indent) + "<rank>");
-        pw1.println(MekHqXmlUtil.indentStr(indent+1)
-                +"<rankNames>"
-                +MekHqXmlUtil.escape(getRankNamesAsString())
-                +"</rankNames>");
-        pw1.println(MekHqXmlUtil.indentStr(indent+1)
-                +"<officer>"
-                +officer
-                +"</officer>");
-        pw1.println(MekHqXmlUtil.indentStr(indent+1)
-                +"<payMultiplier>"
-                +payMultiplier
-                +"</payMultiplier>");
-        pw1.print(MekHqXmlUtil.indentStr(indent) + "</rank>");
+    //region Boolean Comparison Methods
+    public boolean isEmpty(final Profession profession) {
+        return !getRankNames().containsKey(profession) || getRankNames().get(profession).isBlank()
+                || getRankNames().get(profession).equals("-");
     }
 
-    public static Rank generateInstanceFromXML(Node wn) {
-        final String METHOD_NAME = "generateInstanceFromXML(Node)"; //$NON-NLS-1$
+    public boolean indicatesAlternativeSystem(final Profession profession) {
+        return getRankNames().containsKey(profession) && getRankNames().get(profession).startsWith("--");
```

Comment:
Is the "--" encountered anywhere else?

Reply:
It is not, as I used the full strings otherwise.